crates/ironrdp-cliprdr-format/src/bitmap.rs

```diff
@@ -435,6 +428,29 @@ fn validate_v1_header(header: &BitmapInfoHeader) -> Result<(), BitmapError> {
 fn validate_v5_header(header: &BitmapV5Header) -> Result<(), BitmapError> {
     validate_v1_header(&header.header_v1)?;
 
+    // We support only uncompressed DIB bitmaps as it is the most common case for clipboard-copied bitmaps.
+    const DIBV5_SUPPORTED_COMPRESSION: &[BitmapCompression] = &[BitmapCompression::RGB, BitmapCompression::BITFIELDS];
+
+    if !DIBV5_SUPPORTED_COMPRESSION.contains(&header.header_v1.compression) {
+        return Err(BitmapError::Unsupported("unsupported compression"));
+    }
+
+    if header.header_v1.compression == BitmapCompression::BITFIELDS {
+        // Currently, we only support the standard order, BGRA, for the bitfields compression.
+        // Note: we are making sure to interpret the masks as big-endian using `u32::to_be`.
+        // (`to_be` is a no-op on big endian architectures.)
+        let is_bgra = header.red_mask.to_be() == 0x00FF0000
+            || header.green_mask.to_be() == 0x0000FF00
+            || header.blue_mask.to_be() == 0x000000FF
+            || header.alpha_mask.to_be() == 0xFF000000;
```

Comment:
Although we may never hit this problem in practice, I ensured proper interpretation across all native endianness using `to_be`

Reply:
`to_be` will actually break the code here, constants on the `rhs` (e.g. `0x00FF0000`) are implicit `le` on most of the systems, but could be `be` in some bizarre cases. However, the left-hand side is always represented as `be` because of `to_be`. 

AFIK, numeric values in our PDUs have values with endianness native to the current machine, therefore comparison without `to_le`/`to_be` is fine

Reply:
I meant to use `to_le` in this case. This is fixed in https://github.com/Devolutions/IronRDP/pull/286/commits/76379ab2600e7a65edca5c188150de9957f5a9c0

`to_le` is a no-op when little-endian is the native endianness, so it’s basically free on most systems. However, on a big endian architecture, the code is wrong `to_le`, because the hexadecimal literal will not be converted to big endian automatically when compiling for such target. (Albeit, I don’t expect to see such target in practice.)

Reply:
Update: my assumption was wrong.

```rust
fn main() {
    let network_packet = [0xFF, 0x00];
    let int = u16::from_le_bytes(network_packet);
    println!("{:02X?}", int.to_ne_bytes());
    assert_eq!(int, 0x00FF);
}
```

![image](https://github.com/Devolutions/IronRDP/assets/3809077/2861553d-f1cd-42af-af5c-5244c3e27c48)

As you said!